core/client/src/notifications.rs

```diff
@@ -147,32 +208,84 @@ impl<Block: BlockT> StorageNotifications<Block> {
 						}
 					}
 				},
+			};
+			if let Some(child_filters) = child_filters {
+				for (c_key, filters) in child_filters.into_iter() {
+					if let Some((listeners, wildcards)) = self.child_listeners.get_mut(&c_key) {
+						if let Some(filters) = filters {
+							for key in filters.into_iter() {
+								let remove_key = match self.listeners.get_mut(&key) {
```

Comment:
I believe this should be `listeners` not `self.listeners`
```suggestion
								let remove_key = match listeners.get_mut(&key) {
```

worth a test as well